docs/usage.md

```diff
@@ -400,3 +401,44 @@ Multiple package name to module name mappings are joined by a comma (`,`):
 ```shell
 deptry . --package-module-name-map 'foo-python=foo,bar-python=bar'
 ```
+
+#### PEP 621 dev dependency groups
+
+PEP 621 does [not define](https://peps.python.org/pep-0621/#recommend-that-tools-put-development-related-dependencies-into-a-dev-extra) a standard convention for specifying development dependencies. However, deptry offers a mechanism to interpret specific optional dependency groups as development dependencies.
+
+By default, all dependencies under `[project.dependencies]` and `[project.optional-dependencies]` are extracted as regular dependencies. By using the `--pep621-dev-dependency-groups` argument, users can specify which groups defined under `[project.optional-dependencies]` should be treated as development dependencies instead. This is particularly useful for projects that adhere to PEP 621 but do not employ a separate build tool for declaring development dependencies.
```

Comment:
I don't think that `-dev-` is the right word. I think that `-optional-` is.

For example, if we look at the [Setuptools documentation for `optional-dependencies`](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies), we see a user-facing (not "dev" facing) use case:

> A use case for this approach is that other package can use this “extra” for their own dependencies. For example, if Package-B needs Package-A with PDF support installed, it might declare the dependency like this:

Reply:
Ah, I now get it! You can probably ignore this comment.